packages/engine/Source/Scene/ArcGisMapServerImageryProvider.js

```diff
@@ -185,6 +194,72 @@ function ArcGisMapServerImageryProvider(options) {
   const resource = Resource.createIfNeeded(options.url);
   resource.appendForwardSlash();
 
+  let accessToken;
+  switch (options.basemap_id) {
+    case "satellite":
+      {
+        accessToken = defaultValue(
+          options.token,
+          ArcGISMapServiceWorldImagery.defaultAccessToken
+        );
+        const server = Resource.createIfNeeded(
```

Comment:
I don't believe the value of `server` is actually being used. 

It's also not clear that the intention is that setting `options.basemapId` will override the `options.url` parameter. 

Perhaps instead a factory function would make determining the URL from a particular type a bit more straightforward:
```
ArcGisMapServerImageryProvider.fromBasemapType = function (basemapType, options) {
    let server;
    switch (basemapType) {
       // ...
    }

   return new ArcGisMapServerImageryProvider({
       ...options,
       url: server,
   });
}
```

This is similar to the `IonResource.fromAssetId` factory functions.

The usage then becomes:

```
const viewer = new Cesium.Viewer("cesiumContainer", {
  imageryProvider: Cesium.ArcGisMapServerImageryProvider.fromBasemapType({
     ArcGisBaseMapType.SATELLITE,
    //token: "<token>",
  }),
});
```